### ISSUE TITLE
Unable to setup Docker in elementary OS

### ISSUE THREAD

**Issue by fmoliveira** [docker/docker#14999]
I'm following the installation steps for Linux from official website at http://docs.docker.com/linux/step_one/ but when I run the script from `get.docker.com` I've got a message telling me that my platform is not supported by the installer script.

My distribution is elementary OS 0.3 (freya), which is forked from Ubuntu 14.04 (trusty).

```
$ wget -qO- https://get.docker.com/ | sh

  Either your platform is not easily detectable, is not supported by this
  installer script (yet - PRs welcome! [hack/install.sh]), or does not yet have
  a package for Docker.  Please visit the following URL for more detailed
  installation instructions:

    https://docs.docker.com/en/latest/installation/
```

Surely I can add the APT repository manually to setup Docker, but it'll be good if the official installer steps works in elementary OS, as it's getting more and more popular everyday.

**Comment by fmoliveira** [comment]
Hi @GordonTheTurtle , I've searched for that first. I've found an open PR at #14957 and it'll surely not work because it's just accepting the elementary OS distro but not mapping the repository links to the corresponding Ubuntu ones.

I've made PR #15000 with the fix for this issue. I'll reference #14957 with my PR too.

**Comment by fmoliveira** [comment]
### BUG REPORT INFORMATION

docker version: 1.7.1, build 786b29d
docker info: N/A
uname -a: Linux fmoliveira 3.16.0-34-generic #45~14.04.1-Ubuntu SMP Tue Mar 24 11:14:29 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

Provide additional environment details (AWS, VirtualBox, physical, etc.): Physical elementary OS 0.3 Linux setup.

List the steps to reproduce the issue:
1. Go to http://docs.docker.com/linux/step_one/
2. Follow the installation steps
3. In third step which you'll execute the command `$ wget -qO- https://get.docker.com/ | sh`, the following error message will be given:
```
$ wget -qO- https://get.docker.com/ | sh

  Either your platform is not easily detectable, is not supported by this
  installer script (yet - PRs welcome! [hack/install.sh]), or does not yet have
  a package for Docker.  Please visit the following URL for more detailed
  installation instructions:

    https://docs.docker.com/en/latest/installation/
```

Describe the results you received: I'm unable to setup Docker through the official installer script in my Linux distribution.

Describe the results you expected: I wanted to be able to setup Docker through the official installer script in my Linux distribution.

Provide additional info you think is important: I'm using a Linux distribution which is forked from Ubuntu but not as popular as that. It's very easy to add support for my distribution because it's just a matter of mapping it to the corresponding Ubuntu version.

**Comment by fmoliveira** [comment]
I'm sorry. I've fixed the issue according to the guidelines.

**Comment by fmoliveira** [comment]
Pull request #15000 will fix this issue.

**Comment by unclejack** [comment]
Distributions such as this one should really stop messing with the lsb-release. They're not really different distributions, they're all based on Ubuntu 14.04, they add some extra repositories and install some more packages.

Not messing with lsb-release would help a lot with not breaking all the software which checks lsb-release to see if it's a supported distribution.

**Comment by fmoliveira** [comment]
Hi @unclejack ,

Thanks a lot for your thoughts. I agree that your concerns are valid, and I've tried hard to avoid breaking the script for any other distros.

It's exactly because these distros are all based in Ubuntu that I think we should want them to use Ubuntu repositories for installing Docker.

The fix from #15001 will first check if the `-u` option is supported to avoid breaking the whole script. With the proper checks to avoid breaking the script, this seems to be a better practice than to hard code corresponding Ubuntu versions in the script.

If the `-u` option is supported, Docker will be installed from corresponding Ubuntu repository. If not, the LSB will not be queried. The script have been tested in a few different major and forked distros.

It'd be great if you could join the discussion in #15001 to provide your feedback in code review.

Thanks a lot for for your thoughts on this. :+1:

**Comment by fmoliveira** [comment]
Closing in favour of #15001 .
